README.md

```diff
@@ -1,6 +1,36 @@
 # transmogrifier
 
-Data transformation CLI app for TIMDEX
+An application to transform source records to the TIMDEX data model to facilitate ingest into an OpenSearch index.
+
+## Description
+
+TIMDEX ingests records from various sources with different metadata formats, necessitating an application to transform those source records to a common metadata format. This application processes both XML and JSON source records and outputs a JSON file of records formatted according to the TIMDEX data model. 
+
+```mermaid
+---
+title: Transmogrifier in the TIMDEX pipeline
+---
+flowchart TD
+    ArchivesSpace
+    DSpace
+    GeoData
+    MARC
+    transmogrifier((transmogrifier))
+    JSON
+    timdex-index-manager
+    ArchivesSpace[("ArchivesSpace\n(EAD XML)")] --> transmogrifier
+    DSpace[("DSpace\n(METS XML)")] --> transmogrifier
+    GeoData[("GeoData\n(Aardvark JSON)")] --> transmogrifier
+    MARC[("Alma\n(MARCXML)")] --> transmogrifier
+    transmogrifier --> JSON["TIMDEX JSON"]
+    JSON[TIMDEX JSON file] --> timdex-index-manager((timdex-index-manager))
+```
+
+The TIMDEX data model is designed to produce records that can be successfully ingested into an OpenSearch index and contains data fields that are broadly applicable to various types of records. `transmogrifier` contains different validators to ensure that the record is structured properly and that certain types of values, such as dates, align with OpenSearch's expectations.
+
+Each source is defined with configuration values and a dedicated transform class to process records from that source. For each transform class, various errors and warnings are logged. Some errors are logged and the entire source record is skipped because it is incorrectly formatted, while others are merely logged as warnings for later review. The application also determines which records are marked as deleted in each source and removes those record from the OpenSearch index. 
```

Comment:
This is some nice writing.  It's covering a _lot_ of ground, but it's clear and concise and easy to follow.  Nice work!  Maybe additional sections or docs are added later, or links out, but this is a great overview of this codebase right in front of you.

Reply:
One suggestion: I wonder if it'd be helpful to say something like "it is skipped because it is does not meet minimum TIMDEX record requirements" instead of "it is incorrectly formatted"?

Thinking about recent discussions leveraging the distinction between format and data.  If a record is fully missing a field, not sure I'd consider that formatting, it feels like it's missing required data.  Maybe just semantics but feels like an opportunity to seed the reader with a question like, "well, where _are_ the TIMDEX data model requirements?"  and not just assume it's a bad XML angle bracket or something that causes records to get skipped.

Reply:
I think it's actually implying that something is seriously wrong with the source record, such as EAD without an `archdesc` element or MARC without an `008` or leader field. I can definitely phrase it better but it's closer to bad angle bracket in the source than not having a required field, if that makes sense

Reply:
Maybe `Some errors are logged and the entire source record is skipped because the severity implies it should not be processed until fixed, while others are merely logged as warnings for later review.`?

Reply:
> Maybe `Some errors are logged and the entire source record is skipped because the severity implies it should not be processed until fixed, while others are merely logged as warnings for later review.`?

I think that's stellar.  Thanks for the update.  I think the "severity" word is key, as it bypasses tricky conversations of "format" vs "data", technically invalid XML vs XML that doesn't have arbitrarily required fields by us.  Looks good to me!